### PR TITLE
bugfix: rephrase error messages (#193)

### LINKED ISSUE #193: Messages aren't easy to understand
Hello,

I think messages returned for failed cases aren't very easy.

Here is an example:

```java
  @Property
  public void firstTest(final List<Integer>values) {
    final ExerciseOne instance = new ExerciseOne();
    Assert.assertTrue(instance.absoluteAverage(values) >= 0);
  }
```

Here is the message I get:

```java
java.lang.AssertionError: Property firstTest falsified via shrinking: null
Shrunken args: [[-1209279219, 938232675]]
Original failure message: [null]
Original args: [[-1209279219, -1986811869]]
Seeds: [7190051195178127153]
```

Here, I thought that the failing case is the input `null`, but it's not the case, as this check is manually checked in another case.

Instead, I feel like it's "[-1209279219, 938232675]", even though I thought the lib tried to provide a list of list of integers at first (it's an overflow issue here).

Could the failing case be clearer from the log messages?

For example, what about:

``` java
java.lang.AssertionError: Property named 'firstTest' failed with the following shrunk arguments: [-1209279219, 938232675]
First value found to also provoke a failure: [-1209279219, -1986811869]
Seeds: [7190051195178127153]
```

Version tested: 0.7 & 0.8-beta-2.

### PR DESCRIPTION
**Pull request by Jiehong**
In order to fix #193 I opened, I propose the following formatting changes.

Hopefully, it makes some sense :)

Let me know if you want any changes!

### CODE REVIEW

File: core/src/main/java/com/pholser/junit/quickcheck/runner/PropertyFalsified.java [hunk 1]
```diff
@@ -37,14 +37,16 @@ static AssertionError counterexampleFound(
         Object[] args,
```
Comment (pholser) [inline core/src/main/java/com/pholser/junit/quickcheck/runner/PropertyFalsified.java]: I'd prefer you not to declare locals as `final`.

File: core/src/main/java/com/pholser/junit/quickcheck/runner/PropertyFalsified.java [hunk 2]
```diff
@@ -37,14 +37,16 @@ static AssertionError counterexampleFound(
...
         long[] seeds,
         AssertionError e) {
```
Comment (pholser) [inline core/src/main/java/com/pholser/junit/quickcheck/runner/PropertyFalsified.java]: Add a blank line after the arg list.

File: core/src/test/java/com/pholser/junit/quickcheck/runner/PropertyFalsifiedUtilityClassTest.java [hunk 3]
```diff
@@ -26,11 +26,119 @@ a copy of this software and associated documentation files (the
 package com.pholser.junit.quickcheck.runner;
 
 import com.pholser.junit.quickcheck.UtilityClassesUninstantiabilityHarness;
```
Comment (pholser) [inline core/src/test/java/com/pholser/junit/quickcheck/runner/PropertyFalsifiedUtilityClassTest.java]: I don't use `*` imports in the project except in `static` imports. Please expand this out.

File: core/src/test/java/com/pholser/junit/quickcheck/runner/PropertyFalsifiedUtilityClassTest.java [hunk 4]
```diff
@@ -26,11 +26,119 @@ a copy of this software and associated documentation files (the
...
+import org.junit.*;
+
 
 public class PropertyFalsifiedUtilityClassTest
     extends UtilityClassesUninstantiabilityHarness {
 
     public PropertyFalsifiedUtilityClassTest() {
         super(PropertyFalsified.class);
     }
+
```
Comment (pholser) [inline core/src/test/java/com/pholser/junit/quickcheck/runner/PropertyFalsifiedUtilityClassTest.java]: A couple of things on these tests:

1) Please static-import `assertEquals`.

2) Can you separate the tests into "arrange", "act", and "assert" sections with single blank lines, as described [here](http://wiki.c2.com/?ArrangeActAssert)?

File: core/src/test/java/com/pholser/junit/quickcheck/runner/PropertyFalsifiedUtilityClassTest.java [hunk 5]
```diff
@@ -26,11 +26,119 @@ a copy of this software and associated documentation files (the
...
+            arguments,
+            seeds,
+            smallerFailure,
+            originalFailure);
+        final String expected = "Property named 'mySuperProperty' failed:\n"
+            + "With arguments: [first]\n"
+            + "First arguments found to also provoke a failure: [first, second, third]\n"
+            + "Seeds for reproduction: [12345, 8842]";
+        Assert.assertEquals(expected, actual.getMessage());
+    }
```
Comment (pholser) [inline core/src/test/java/com/pholser/junit/quickcheck/runner/PropertyFalsifiedUtilityClassTest.java]: Can you modify src/site/markdown/usage/seed.md to reflect the new format of the failure message?

File: core/src/main/java/com/pholser/junit/quickcheck/runner/PropertyFalsified.java (hunk 1 above)
Comment (Jiehong) [reply core/src/main/java/com/pholser/junit/quickcheck/runner/PropertyFalsified.java]: sure, but would you mind giving some details as to why you prefer this?

Comment (pholser) [reply core/src/main/java/com/pholser/junit/quickcheck/runner/PropertyFalsified.java]: Certainly! I find that `final`-izing local vars and method parameters introduces noise into the code, and provides a benefit that I don't find very helpful (assurance that the values in question won't change).

When coming into another code base that opts for marking these vars `final`, I'll gladly oblige. In junit-quickcheck and other personal projects of mine, I lean toward avoiding these declarations. I would likely lean on static analysis tools like FindBugs or Checkstyle to warn me of potentially confusing constructs where a local is getting modified a lot.

**Pull request merged**